pkg/controller/vitesscell/reconcile_vtgate.go

```diff
@@ -161,6 +161,12 @@ func (r *ReconcileVitessCell) reconcileVtgate(ctx context.Context, vtc *planetsc
 			curObj := obj.(*appsv1.Deployment)
 
 			status := &vtc.Status.Gateway
+			if replicas :=curObj.Spec.Replicas; replicas != nil {
+				status.Replicas = *replicas
+			}
+			if selector := curObj.Spec.Selector; selector != nil {
+				status.LabelSelector = selector.String()
+			}
```

Comment:
The String() function already checks for nil receiver, so I think it is safe to replace this with
```go
status.LabelSelector = curObj.Spec.Selector.String()
```